src/jdk.incubator.concurrent/share/classes/jdk/incubator/concurrent/ScopedValue.java

```diff
@@ -43,71 +42,109 @@
- * There may be cases when an operation might need to use the same scoped value
- * to communicate a different value to the methods that it calls. The requirement is not
- * to change the original binding but to establish a new binding for nested calls. If an
- * scoped value already has a value, then {@code run} or {@code call} methods may be
- * invoked to run another operation with a newly-bound value. Code executed by the
- * operation will read the new value of the scoped value. The scoped value reverts to its
- * previous value when the operation completes.
+ * <p> Consider the following example with a scoped value {@code USERNAME} that is
+ * <em>bound</em> to the value "{@code duke}" for the execution, by a thread, of a run
+ * method that invokes {@code doSomething()}.
+ * {@snippet lang=java :
+ *     // @link substring="newInstance" target="#newInstance" :
+ *     private static final ScopeValue<String> USERNAME = ScopeValue.newInstance();
+ *
+ *     ScopedValue.where(USERNAME, "duke", () -> doSomething());
+ * }
+ * Code executed directly or indirectly by {@code doSomething()} that invokes {@code
+ * USERNAME.get()} will read the value "{@code duke}". The scoped
+ * value is bound when executing {@code doSomething()} and becomes unbound when {@code
+ * doSomething()} completes (normally or with an exception). If one thread were to call
+ * {@code doSomething()} with {@code USERNAME} bound to "{@code duke1}", and another
+ * thread were to call the method with {@code USERNAME} bound to "{@code duke2}", then
+ * {@code USERNAME.get()} would read the value "{@code duke1}" or "{@code duke2}",
+ * depending on which thread is executing.
+ *
+ * <p> In addition to the {@code where} method that executes a {@code run} method, {@code
+ * ScopeLocal} defines the {@link #where(ScopedValue, Object, Callable)} method to execute
+ * a method that returns a result. It also defines the {@link #where(ScopeValue, Object)}
+ * method for cases where it is useful to accumulate mappings of {@code ScopeValue} to
+ * value.
  *
- * <h2> Sharing scoped values across threads </h2>
+ * <p> A {@code ScopedValue} will typically be declared in a {@code final} and {@code
+ * static} field. The accessibility of the field will determine which components can
+ * bind or read its value.
  *
- * Scoped-value bindings can be shared across threads when used in conjunction with
- * {@link StructuredTaskScope}. Creating a {@code StructuredTaskScope} captures the
- * current thread's scoped-value bindings for inheritance by threads {@link
- * StructuredTaskScope#fork(Callable) forked} in the task scope. This means that a thread
- * may bind a scoped value and share its value in a structured concurrency
- * context. Threads forked in the task scope that read the scoped value will read
- * the value bound by the thread that created the task scope.
+ * <p> The value of a {@code ScopedValue} should be an immutable object. This is
```

Comment:
This statement seems a bit strong. A large part of the usages need the value of a `ScopedValue` to be mutable: object pools, output streams, sockets...

Reply:
Fair point, better to drop it to avoid any confusion.